lib/question.js

```diff
@@ -1406,9 +1411,10 @@ module.exports = {
                     // Render the submission panel
 
                     // If the submission wasn't graded yet, return null to indicate that
-                    // the panel shouldn't be rerendered.
-                    if (submission.graded_at === null) {
-                        callback(null);
+                    // the panel shouldn't be rerendered.  Check if 'feedback' is set because
+                    // that will indicate the grading code was run (even if it is ungradable)
```

Comment:
This feels a bit too sneaky to me. Should we be adding a new property like `grading_succeeded_at` that's separate to `graded_at`, so `graded_at` can mean "ran the grading code"?

Or we could add a new property like `grading_code_run_at` to indicate that we ran the grading code, even if `graded_at = NULL`.

But maybe I'm misunderstanding what's going on here. Adding a whole other DB column just for one check seems pretty heavy-weight.

Reply:
The question is, is this check even needed?  This line used to be `callback(null)` (it did not `return`), so it would still render anyway if it got to this point.  I'll experiment with dropping the check and seeing if anything breaks.

Reply:
Removed, seems to be unnecessary.